### ISSUE TITLE
ServerWebSocket is not able to process further connection messages

### ISSUE THREAD

**Issue by vietj** [eclipse-vertx/vert.x#2879]
Currently server WebSocket handshake will send an HttpResponse message directly leaving the Http1xServerConnection#responseInProgress field with the current response. This prevent the HTTP server request handler to further process any HTTP message such as writability.

As side effect, this prevent the `ServerWebSocket` drain handler to be called, since the response in progress will handle this message.

The `WebSocketServerHandshaker` will take care of sending the `HttpResponse` message so there is no way to use `HttpServerResponse` instead. So after a successful WebSocket handshake the `Http1xServerConnection#responseComplete()` method should be called to signal the response has ended.

**Comment by kingguppy** [comment]
Thanks very much for the quick response. I can confirm that with 3.7.0-SNAPSHOT as of this writing, the reproducer in vert-x3/vertx-web#1200 now performs as expected (showing "Received: 1000" as the last line), and that my actual application is also working as expected so far. I tried with both ServerWebSocket and SockJSSocket in an application-specific load test involving hundreds of clients simultaneously connected and sending/receiving variable amounts of data, and disconnecting/reconnecting. A few https requests were thrown into the mix too.

Is there any chance that if I made a pull request for a backport of this change to the 3.6 branch, it could be released this week? I don't specifically care about the 3.6 branch, just the timeframe for getting a release version - I assume 3.7.0 will take longer. I don't mean to nag, just to check :)
